application.py

```diff
@@ -204,16 +204,30 @@ def handleUSECluster3():
         raise BeagleError(errors.TOO_FEW_QUESTIONS) #Communicate we don't support
     
     embeddings, data_used_for_demo, q_ids_list = get_data_embeddings(data)
+    print("q_ids_list:",q_ids_list)
+    output = []
     if(len(data_used_for_demo)<50):
         
         best_scores = list(map(int,best_score_HAC_sparse(embeddings, data_used_for_demo, 2)[1]))
-        return jsonify(return_cluster_dict(best_scores,q_ids_list))
+        final_list = return_cluster_dict(best_scores,q_ids_list)
+        labels = return_cluster_labels_nmi_ngrams_centroid(embeddings, data_used_for_demo, q_ids_list, final_list)
+        for cluster_id in final_list:
+            q_ids = final_list[cluster_id]
+            cluster_label = ' , '.join(labels[cluster_id])
+            output.append({"q_ids":q_ids, "label":cluster_label})
+        return jsonify(output)
     
         
     else:
         
         best_scores = list(map(int,get_best_HAC_normal(embeddings, data_used_for_demo)[1]))
-        return jsonify(return_cluster_dict(best_scores,q_ids_list))
+        final_list = return_cluster_dict(best_scores,q_ids_list)
```

Comment:
I think it'd be good to rename "final_list" to make it clear it is a list of clusters. I'd use something like "cluster_list"

Reply:
Yes makes sense :D